drivers/flash/Kconfig.emulator

```diff
@@ -0,0 +1,133 @@
+# Kconfig - Flash driver emulator config
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if FLASH && FLASH_EMULATOR
```

Comment:
Since `FLASH_EMULATOR` already depends on `FLASH`, this could just be `if FLASH_EMULATOR`.